cf_remote/commands.py

```diff
@@ -475,18 +475,56 @@ def uninstall(hosts):
         errors += uninstall_host(host)
     return errors
 
-def deploy(hubs, directory):
-    assert(directory.endswith("/masterfiles"))
-    assert(os.path.isfile(directory + "/autogen.sh"))
+def deploy_tarball(hubs, tarball):
+    assert os.path.isfile(tarball)
+
+    if not tarball.endswith((".tgz", ".tar.gz")):
+        log.error("The masterfiles directory must be in a gzipped tarball (.tgz or .tar.gz)")
+        return 1
+
+    errors = 0
+    for hub in hubs:
+        errors += deploy_masterfiles(hub, tarball)
+    return errors
+
+def deploy(hubs, masterfiles):
+    if masterfiles.startswith(("http://", "https://")):
+        urls = [masterfiles]
+        paths = _download_urls(urls)
+        assert len(paths) == 1
+        masterfiles = paths[0]
+        log.debug(f"Deploying downloaded: {masterfiles}")
+    else:
+        masterfiles = os.path.abspath(os.path.expanduser(masterfiles))
+        log.debug(f"Deploy path expanded to: {masterfiles}")
+
+    if masterfiles.endswith("/"):
```

Comment:
yes, this is the  bit I wanted changed recently. :+1: but which even the previous business which expands the absolute path might take care of. :)